### ISSUE TITLE
expect(map).to.be.empty does not behave naturally

### ISSUE THREAD

**Issue by janakdr** [cypress-io/cypress#6072]
```js
it('expect map', () => {
    expect(new Map([['a', 1]])).to.not.be.empty;
});
```

This test fails with the incorrect assertion: "expected {} not to be empty". It seems like Cypress is interpreting the Map object as some kind of DOM element, which it thinks is empty? Similarly, this test incorrectly passes if the `.not` is removed.

<img width="706" alt="Screen Shot 2019-12-31 at 1 48 00 PM" src="https://user-images.githubusercontent.com/10134896/71631017-5659ac80-2bd4-11ea-8dcb-78106d126292.png">

Cypress doesn't explicitly say what it does for the `.empty` predicate in its [assertions documentation](https://docs.cypress.io/guides/references/assertions.html#Chai) but it does link to the [full list of assertions](https://www.chaijs.com/api/bdd/#method_empty) which specify that for a Map, it will assert that the map has size 0. Even if the test code is changed to `expect(new Map([['a', 1]])).to.be.a('map').that.is.not.empty` the "empty" assertion is the one that fails.

**Comment by jennifer-shehane** [comment]
This comes directly from Chai assertions and was not supported until Chai 4.0 as noted here: https://github.com/chaijs/chai/issues/781

Support for asserting `empty` against Map will come in Cypress v4.0 which includes an update of Chai to 4.2.0 and other breaking changes. https://github.com/cypress-io/cypress/issues/4270

The code for this is done in https://github.com/cypress-io/cypress/issues/4270, but this has yet to be released. We'll update this issue and reference the [changelog](https://on.cypress.io/changelog) when it's released.

**Comment by flotwig** [comment]
Released in [`4.0.0`](https://on.cypress.io/changelog#4-0-0).
